src/core/ParticleIterator.hpp

```diff
@@ -16,6 +16,10 @@ struct ParticleIterator : public boost::iterator_facade<
     }
   }
 
+  operator Particle *() const {
```

Comment:
Why should the iterator behave like a pointer to `Particle`?  Usually you dereference an iterator to get the contents.